File: CarHomeWidget/src/main/java/com/anod/car/home/appscache/AppsCacheActivity.java
package com.anod.car.home.appscache;

import com.anod.car.home.app.AppsListActivity;
import com.anod.car.home.model.AppsList;

import android.content.Loader;
import android.os.Bundle;

import java.util.ArrayList;

public abstract class AppsCacheActivity extends AppsListActivity
        implements AppsCacheLoader.Callback {


    @Override
    public Loader<ArrayList<AppsList.Entry>> onCreateLoader(int id, Bundle args) {
        return new AppsCacheLoader(this, this, mAppsList);
    }


}

File: CarHomeWidget/src/main/java/com/anod/car/home/appscache/AppsCacheActivity.kt
package com.anod.car.home.appscache

import android.content.Intent
import com.anod.car.home.app.AppsListActivity
import android.os.Bundle

abstract class AppsCacheActivity : AppsListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel.loaderFactory = { context, callback -> AppsCacheLoader(context, createQueryIntent(), callback) }
    }

    abstract fun createQueryIntent(): Intent
}